e2e/elasticsearch.go

```diff
@@ -256,3 +257,48 @@ func WaitForElasticsearchFromHostPort(host string, port int, maxTimeoutMinutes t
 
 	return true, nil
 }
+
+// WaitForNumberOfHits waits for an elasticsearch query to return more than a number of hits,
+// returning false if the query does not reach that number in a defined number of time.
+func WaitForNumberOfHits(indexName string, query map[string]interface{}, desiredHits int, maxTimeout time.Duration) (SearchResult, error) {
```

Comment:
this is fabulous.  I wonder if (in a coming PR) the next step will be to parametrize this further and make it even more re-usable than it is.  I see we have the index, and # of hits, timeout, and the query.  

It may not be easy or quite appropriate, but I expect keeping fewer copies of the big 'query' listed in code the better... we can chat it, and its certainly mergeable as is, I'm just thinking long term (the first thing someone will do is another query check for something different in ES I'm betting!)